### PR TITLE
Cleanup /query endpoint and use predict default vectorset

### PR DESCRIPTION
**Pull request by jotare**
### Description
Describe the proposed changes made in this PR.

### How was this PR tested?
Describe how you tested this PR.

### CODE REVIEW

File: nucliadb/src/nucliadb/search/search/query.py [hunk 1]
```diff
@@ -463,7 +483,8 @@ async def parse_vector_search(self, request: nodereader_pb2.SearchRequest) -> bo
                         else:
                             incomplete = True
                     else:
-                        query_vector = query_info.sentence.data
+                        for query_vector in query_info.sentence.vectors.values():
```
Comment (lferran) [inline nucliadb/src/nucliadb/search/search/query.py:486]: maybe it's better to use popitem? I know it shouldn't but what if for some reason vectors is empty?

File: nucliadb/src/nucliadb/search/search/query.py [hunk 2]
```diff
@@ -403,38 +437,24 @@ def parse_paragraph_search(self, request: nodereader_pb2.SearchRequest) -> None:
...
+
+        # validate vectorset
+        async with datamanagers.with_ro_transaction() as txn:
+            if not await datamanagers.vectorsets.exists(
+                txn, kbid=self.kbid, vectorset_id=self.vectorset
+            ):
+                raise InvalidQueryError(
+                    "vectorset",
+                    f"Vectorset {self.vectorset} doesn't exist in you Knowledge Box",
+                )
```
Comment (lferran) [inline nucliadb/src/nucliadb/search/search/query.py:456]: this could be hard cached no?

File: nucliadb/src/nucliadb/search/search/query.py (hunk 1 above)
Comment (jotare) [reply nucliadb/src/nucliadb/search/search/query.py:486]: I'd prefer not to modify the `QueryInfo` payload, but I can check for empty vector

File: nucliadb/src/nucliadb/search/search/query.py [hunk 3]
```diff
@@ -382,12 +409,19 @@ async def parse_min_score(self, request: nodereader_pb2.SearchRequest, incomplet
...
+                else:
+                    logger.warning(
+                        "Semantic threshold not found in query information, using default",
+                        extra={"kbid": self.kbid},
+                    )
             else:
                 logger.warning(
-                    "Semantic threshold not found in query information, using default",
-                    extra={"kbid": self.kbid},
+                    "Vectorset unset, using default semantic threshold", extra={"kbid": self.kbid}
```
Comment (lferran) [inline nucliadb/src/nucliadb/search/search/query.py:424]: will this log be very chatty? I assume lots of requests won't have the vectorset param provided and will just use the default no?

File: nucliadb/src/nucliadb/search/search/query.py [hunk 4]
```diff
@@ -191,18 +191,45 @@ def _get_query_information(self) -> Awaitable[QueryInfo]:
...
+
+        query_information = await self._get_query_information()
+
+        if query_information.sentence is None:
+            logger.error(
+                "Asking for a vectorset but /query didn't return one", extra={"kbid": self.kbid}
+            )
+            return self.vectorset
+
+        for vectorset in query_information.sentence.vectors.keys():
```
Comment (lferran) [inline nucliadb/src/nucliadb/search/search/query.py:218]: nit: is popitem better/safer?

File: nucliadb/src/nucliadb/search/search/query.py (hunk 2 above)
Comment (lferran) [reply nucliadb/src/nucliadb/search/search/query.py:456]: (I mean checking the lookup for if the vectorset exists)

File: nucliadb/src/nucliadb/search/search/query.py (hunk 3 above)
Comment (jotare) [reply nucliadb/src/nucliadb/search/search/query.py:424]: No, this is a log for either predict or the user setting the vectorset. Shouldn't happen. I'll clarify the log

File: nucliadb/src/nucliadb/search/search/query.py (hunk 4 above)
Comment (jotare) [reply nucliadb/src/nucliadb/search/search/query.py:218]: I prefer not to modify the `query_information` object

**Review by javitonino — APPROVED** [review]

**Pull request merged**